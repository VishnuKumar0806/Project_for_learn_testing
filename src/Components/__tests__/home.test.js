import { render, screen, cleanup } from '@testing-library/react';
import redderer from 'react-test-renderer';
import Home from '../home.js';

afterEach(() => {
    cleanup()
});

test('should render completed todo comoponent', () => {
    const todo = { id: 1, name: 'learn DSA', completed: true };
    render(<Home todo={todo}/>);
    const element = screen.getByTestId('test-1');
    expect(element).toBeInTheDocument()
    expect(element).toHaveTextContent("learn DSA")
})

test('should render not completed todo comoponent', () => {
    const todo = { id: 2, name: 'learn cooking', completed: false };
    render(<Home todo={todo}/>);
    const element = screen.getByTestId('test-2');
    expect(element).toBeInTheDocument()
    expect(element).toHaveTextContent("learn cooking")
    expect(element).not.toContainHTML('<strike>')
})

test(('match snapshot'),()=>{
    const todo = { id: 1, name: 'learn DSA', completed: true };
    const tree = redderer.create(<Home todo={todo}/>).toJSON()
    expect(tree).toMatchSnapshot()
})