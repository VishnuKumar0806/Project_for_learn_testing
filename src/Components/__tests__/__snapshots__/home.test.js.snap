// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<center
  data-testid="test-1"
>
  <h1>
    <strike>
      learn DSA
    </strike>
  </h1>
</center>
`;
